src/api.c

```diff
@@ -1534,8 +1539,55 @@ int cgroup_build_tasks_procs_path(char * const path,
 		err = 0;
 		break;
 	case CGROUP_V2:
-		strncat(path, "cgroup.procs", path_sz - strlen(path));
 		err = 0;
+		snprintf(cg_type_path, FILENAME_MAX, "%scgroup.type", path);
```

Comment:
Should we put this reading of `cgroup.type` in its own function?  Seems like this could be helpful for two reasons:
1.  I'm guessing we'll need to read it again elsewhere
2.  It would make it easier to test.  Testing threads in the functional tests might be tedious (or it might not), but it should be trivial to unit test this

Reply:
Agree, will move the reading `cgroup.type` block into a new function and re-spin the patch.  One idea, would be to extend   `process::create_process_in_cgroup()` to create threads using the `threading` module:
```
def create_process_in_cgroup(self, config, controller, cgname,
                                 cgclassify=True, threads=0):
```
if `threads` argument count is more than one, then create `N` threads and write the tid one of them into `cgroup.threads`